devicetypes/smartthings/zwave-rgbw-bulb.src/zwave-rgbw-bulb.groovy

```diff
@@ -0,0 +1,353 @@
+			}
+		}
+	}
+
+	controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range:"(2700..6500)") {
+		state "colorTemperature", action:"color temperature.setColorTemperature"
+	}
+
+	main(["switch"])
+	details(["switch", "levelSliderControl", "rgbSelector", "colorTempSliderControl"])
+}
+
+private getCOLOR_TEMP_MIN() { 2700 }
+private getCOLOR_TEMP_MAX() { 6500 }
+private getWHITE_MIN() { 0 } // min for Z-Wave coldWhite and warmWhite paramaeters
+private getWHITE_MAX() { 255 } // max for Z-Wave coldWhite and warmWhite paramaeters
+private getCOLOR_TEMP_DIFF() { COLOR_TEMP_MAX - COLOR_TEMP_MIN }
+private getRED() { "red" }
+private getGREEN() { "green" }
+private getBLUE() { "blue" }
+private getWARM_WHITE() { "warmWhite" }
+private getCOLD_WHITE() { "coldWhite" }
+private getRGB_NAMES() { [RED, GREEN, BLUE] }
+private getWHITE_NAMES() { [WARM_WHITE, COLD_WHITE] }
+private getCOLOR_NAMES() { RGB_NAMES + WHITE_NAMES }
+
+private def rgbToHSV(red, green, blue) {
+	def hex = colorUtil.rgbToHex(red as int, green as int, blue as int)
+	def hsv = colorUtil.hexToHsv(hex)
+	return [hue: hsv[0], saturation: hsv[1], value: hsv[2]]
+}
+
+private def huesatToRGB(hue, sat) {
+	def color = colorUtil.hsvToHex(Math.round(hue) as int, Math.round(sat) as int)
+	return colorUtil.hexToRgb(color)
+}
+
+private zwaveWhiteToTemp(warmWhite, coldWhite) {
+	warmWhite = warmWhite < WHITE_MIN ? WHITE_MIN : warmWhite > WHITE_MAX ? WHITE_MAX : warmWhite
+	coldWhite = coldWhite < WHITE_MIN ? WHITE_MIN : coldWhite > WHITE_MAX ? WHITE_MAX : coldWhite
+	// Compute temp as concensus between warm white and cold white.
+	def warmTemp = (WHITE_MAX - warmWhite ) * COLOR_TEMP_DIFF / WHITE_MAX + COLOR_TEMP_MIN
+	def coldTemp = coldWhite * COLOR_TEMP_DIFF / WHITE_MAX + COLOR_TEMP_MIN
+	def temp = ((warmTemp + coldTemp) * 0.5) as Integer
+}
+
+private zwaveTempToWarmWhite(temp) {
+	temp = temp < COLOR_TEMP_MIN ? COLOR_TEMP_MIN : temp > COLOR_TEMP_MAX ? COLOR_TEMP_MAX : temp
+	def warmValue = ((COLOR_TEMP_MAX - temp) / COLOR_TEMP_DIFF * WHITE_MAX) as Integer
+}
+
+private zwaveTempToColdWhite(temp) {
+	(WHITE_MAX - zwaveTempToWarmWhite(temp))
+}
+
+private def zwToStColor(red, green, blue, warmWhite, coldWhite) {
+	def stColor = [:]
+	def colors = [red, green, blue]
+	def hexColor = "#" + colors.collect { Integer.toHexString(it).padLeft(2, "0") }.join("")
+	stColor["color"] = hexColor
+	def hsv = rgbToHSV(*colors)
+	stColor["hue"] = hsv.hue
+	stColor["saturation"] = hsv.saturation
+	stColor["colorTemperature"] = zwaveWhiteToTemp(warmWhite, coldWhite);
+	log.debug(stColor)
+	stColor
+}
+
+def updated() {
+	response(refresh())
+}
+
+def installed() {
+	log.debug "installed()..."
+	sendEvent(name: "checkInterval", value: 1860, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "0"])
+	sendEvent(name: "level", value: 100, unit: "%")
```

Comment:
Is this a common thing to set the values to 0 to start out with?

Reply:
Per offline conversation, legacy offlinePingable functionality is now removed from this DTH.